core/src/main/scala/scala/pickling/internal/RuntimePicklerRegistry.scala

```diff
@@ -0,0 +1,61 @@
+package scala.pickling
+package internal
+
+import scala.pickling.runtime.CustomRuntime
+import scala.pickling.spi.PicklerRegistry
+
+/** Helper class to register picklers and unpicklers that
+  * handle special/crazy classes for runtime generation.
+  *
+  * Related to [[scala.pickling.runtime.CustomRuntime]].
+  * Note: Currently this only handles Tuple2s.
+  */
+trait RuntimePicklerRegistry extends CustomRuntime {
```

Comment:
That's very cool! Hope to see something soon :smile: Yet I think we'll need the custom runtime picklers/unpicklers to allow external third-party users to define and use them.